### PR TITLE
Feature/1213 send email verification link

### LINKED ISSUE #1213: Support verification of email addresses when candidates set up new account/change email address DRAFT
## Background
In [this ticket,](https://app.zenhub.com/workspaces/platform-development-5ea838cd2aec471eb6d14139/issues/gh/jac-uk/apply/1212) we are preventing candidates from setting up applications with an email address that differs from that on their profile. Long overdue is a mechanism that verifies an email address when it is set up/changed.

## User Story
As a candidate, I would like my email address to be verified to avoid any issues with typos.

## Feature(s) Description
- [ ] When a new candidate sets up their account for the first time, the platform generates a verification link which is sent to the specified email address via Notify; when the link is clicked, the candidate account (and their status if necessary) is updated in the database
- [ ] When a candidate changes the email address in their profile, the platform generates a verification link which is sent to the specified email address via Notify; when the link is clicked, the candidate account (and their status if necessary) is updated in the database

## Acceptance Criteria 

- [ ] Activation links have an expiry time of ???
- [ ] 

## Definition of Done
- [ ] User stories/acceptance criteria met
- [ ] Internal reviews passed (feedback actioned)
- [ ] User testing passed
- [ ] Relevant technical documentation updated
- [ ] User guidance updated
- [ ] Deployed and merged without errors

## User Testing Steps
Specify for users what is being tested (but not how to test it.)

### PR DESCRIPTION
**Pull request by drieJAC**
### THE FOLLOWING MUST BE DONE BY THE DEV TEAM PRIOR TO UT:

- Merge the following PR on develop: https://github.com/jac-uk/digital-platform/compare/feature/apply-1213-verification-email?expand=1
- The Notification Service must be activated during User Testing

Added an email verification check to both the Sign Up and Change Email processes.
When a user either signs up or changes their email address (on the profile page) they are sent an email with a link in. When they click on the link they are verifying that the email address is theirs. Sign up/Email change processes require this to be done.
The user can also request a new email verification link to be sent to them, eg if they have deleted or not received the original email or if the link has timed out (after 3 days).
The sign up process has been split into two steps. The first step collects the email and password (the user then verifies the email by clicking the link that's sent to them) then they go to the second sign up step where they enter the remaining personal details.

Closes #1213 

## Who should test?
✅ Product owner
✅ Developers
✅ UTG

## How to test?

**IMPORTANT**: To facilitate multiple tests below you will need to sign up multiple times. Each time you need to use a unique email address and have them sent to your inbox. To achieve this using a **SINGLE** email account you can add a plus sign '+' with a number after it.
So if your email address is: **john@judicialappointments.gov.uk**
You can use the following to sign up with (and the emails will all be sent to: john@judicialappointments.gov.uk):

**john+1@judicialappointments.gov.uk
john+2@judicialappointments.gov.uk
john+3@judicialappointments.gov.uk**
etc

**NOTE: It may take up to a minute to receive the email**

Go to the following URL: https://jac-apply-develop--pr1240-feature-1213-send-em-ypawr0ll.web.app/sign-in?nextPage=/vacancies

**Test 1**

1. Click on the 'create an account' link
2. Enter your email and password (remembering to use the plus sign with a number as detailed above) and press the submit button **Note down the email address you have used.**
3. You will receive a message saying you need to go to your inbox and click on the link in the email to verify your email address
4. Go to your inbox and find the email sent. **Check the email address is the same as the one you used in step 2 above!**
5. Click on the link in the email
6. A new window will open in your browser telling you that you have verified your email and giving you a button to press to 'continue'. Press the button.
7. You will be taken to a page confirming you have verified your email address. After a few seconds it will automatically take you to Step 2 of the Sign Up process.
8. Fill in the remaining details and press the submit button
9. This completes registration and you will be taken to the Vacancies page. Make sure you can see the 'Your Profile' link on the left.

**Test 2**
Now that you are logged in you can test the Change Email functionality
1. On the Vacancies page you will see the Your Profile' link on the left.
2. Click on this link and then click on the 'Edit' link (to the right of the 'Your Profile' text
3. Change your email address (remembering to use the plus sign with a number as detailed above) and press the submit button **Note down the email address you have used.**
4. You will receive a message saying you need to go to your inbox and click on the link in the email to verify your email address
5. Go to your inbox and find the email sent. **Check the email address is the same as the one you used in step 2 above!**
6. Click on the link in the email
7. A new window will open in your browser telling you that you have verified your email and giving you a button to press to 'continue'. Press the button.
8. You will be taken to a page confirming you have verified your email address. After a few seconds it will automatically take you to the Vacancies page.
9. Click on the 'My Profile' link on the left again
10. Scroll down to check that the email address in your profile matches your new one

**Test 3**
Stay logged in and perform the following steps
1. On the Vacancies page you will see the Your Profile' link on the left.
2. Click on this link and then click on the 'Edit' link (to the right of the 'Your Profile' text
3. Change your email address (remembering to use the plus sign with a number as detailed above) and press the submit button **Note down the email address you have used.**
4. You will receive a message saying you need to go to your inbox and click on the link in the email to verify your email address
5. Go to your inbox and find the email sent. **Check the email address is the same as the one you used in step 2 above!**
6. Click on the link in the email
7. A new window will open in your browser telling you that you have verified your email and giving you a button to press to 'continue'. Press the button.
8. You will be taken to a page confirming you have verified your email address. After a few seconds it will automatically take you to the Vacancies page. Close this page.
9. Go to the other window/tab you have open in step 4. (This is the page that asked you to go to your inbox to verify your email address)
10. Click on the Refresh button. The 'Your Profile' link should then appear on the left hand side.

**Test 4**
Stay logged in and perform the following steps
1. On the Vacancies page you will see the Your Profile' link on the left.
2. Click on this link and then click on the 'Edit' link (to the right of the 'Your Profile' text
3. Change your email address (remembering to use the plus sign with a number as detailed above) and press the submit button **Note down the email address you have used.**
4. You will receive a message saying you need to go to your inbox and click on the link in the email to verify your email address
5. DO NOT go to your inbox to verify the email address. Instead go to your inbox and DELETE the email sent to you.
6. Go back to the webpage in step 4 (This is the page that asked you to go to your inbox to verify your email address) and click on the Resend button. You should get an error saying 'Too many email verification requests. Please try again later.'.
7. Come back after 30 minutes and press the Resend button again. When it has completed go to your inbox and you should have a new email allowing you to confirm your email address.
8. Click on the link in the email
7. A new window will open in your browser telling you that you have verified your email and giving you a button to press to 'continue'. Press the button.
8. You will be taken to a page confirming you have verified your email address. After a few seconds it will automatically take you to the Vacancies page.
9. Click on the 'My Profile' link on the left again
10. Scroll down to check that the email address in your profile matches your new one

## Risk - how likely is this to impact other areas?
🟠 Medium risk - this does change code that is shared with other areas

---
PREVIEW:DEVELOP
_can be OFF, DEVELOP or STAGING_

### CODE REVIEW

File: src/views/SignUpStep1.vue [hunk 1]
```diff
@@ -100,7 +100,7 @@
             label="National Insurance number"
             hint="It’s on your National Insurance card, payslip or P60. For example, ‘QQ 12 34 56 C’."
             type="nino"
-          />
+          /> -->
 
           <button
```
Comment (HalcyonJAC) [inline src/views/SignUpStep1.vue:105]: Could we use the `ActionButton` here to prevent double-clicking?

Comment (HalcyonJAC) [inline src/views/SignUpStep1.vue:105]: Could we use the `ActionButton` here to prevent double-clicking?

**Comment by HalcyonJAC** [comment]
The page doesn't redirect to the next page. Please see the following video: 

https://github.com/user-attachments/assets/a8af4791-907c-4286-a522-dc82d1399310

**Comment by drieJAC** [comment]
> The page doesn't redirect to the next page. Please see the following video:
> 
>  Screen.Recording.2024-10-14.at.11.43.20.mov

OK I have fixed this now - good spot

**Comment by HalcyonJAC** [comment]
Two minor issues:
1. The `<br>` is rendered.

    image

2. The page of "Create an account - Step 2" flashes and all error messages are gone when I click the "Complete Registration" button. 

    https://github.com/user-attachments/assets/f192e389-b78b-4fc1-b659-8bfcbc4cae85

**Review by HalcyonJAC — APPROVED** [review]

**Review by tomlovesgithub — APPROVED** [review]

**Pull request merged**